app/src/main/java/nmd/nomad/api/ServiceGenerator.java

```diff
@@ -0,0 +1,21 @@
+package nmd.nomad.api;
+
+import com.squareup.okhttp.OkHttpClient;
+
+import retrofit.RestAdapter;
+import retrofit.client.OkClient;
+
+/**
+ * Created by nicolasiensen on 1/28/15.
+ */
+public class ServiceGenerator {
+    public static <S> S createService(Class<S> serviceClass, String baseUrl) {
+        RestAdapter.Builder builder = new RestAdapter.Builder()
+                .setEndpoint(baseUrl)
+                .setClient(new OkClient(new OkHttpClient()));
+
+        RestAdapter adapter = builder.build();
+
+        return adapter.create(serviceClass);
```

Comment:
Hey @intrications,

I agree that we should have only one instance of a service for each API, but ServiceGenerator creates any kind of Retrofit API service and we can't make it a singleton class.

If we change ServiceGenerator to something like GomadServiceGenerator, that creates services only for GomadClient interface, maybe we can make it a singleton class.

Do you understand my point here? Do you agree?